runtime/firebase-storage.js

```diff
@@ -0,0 +1,133 @@
+// @
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+import StorageProviderBase from './storage-provider-base.js';
+import firebase from '../platform/firebase-web.js';
+import assert from '../platform/assert-web.js';
+
+class FirebaseKey {
+  constructor(key) {
+    var parts = key.split('://');
+    this.protocol = parts[0];
+    assert(this.protocol == 'firebase');
+    if (parts[1]) {
+      parts = parts[1].split('/');
+      assert(parts[0].endsWith('.firebaseio.com'));
+      this.databaseUrl = parts[0];
+      this.projectId = this.databaseUrl.split('.')[0];
+      this.apiKey = parts[1];
+      this.location = parts.slice(2).join('/');
+    } else {
+      this.databaseUrl = undefined;
+      this.projectId = undefined;
+      this.apiKey = undefined;
+      this.location = "";
+    }
+  }
+
+  toString() {
+    if (this.databaseUrl && this.apiKey)
+      return `${this.protocol}://${this.databaseUrl}/${this.apiKey}`;
+    return `${this.protocol}://`;
+  }
+}
+
+export class FirebaseStorage {
```

Comment:
Are we planning to export other things, otherwise just export default.

Reply:
Done.